### ISSUE TITLE
Logging warning in "ondemand" GraphQL v4 schema build

### ISSUE THREAD

**Issue by maxime-rainville** [silverstripe/silverstripe-graphql#446]
If you're `.graphql-generated` folder does not exist when you try calling `admin/graphql`, Silverstripe will attempt to regenerate it "on demand". That works pretty well, except for the fact that the output that's meant to go to the console ends up in the HTTP Response.

```
[INFO]: Creating new schema<br>[INFO]: Migrating new schema<br>[INFO]: Deleting temp schema<br>[INFO]: Total types: 59<br>[INFO]: Types built: 59<br>[INFO]: *CopyToStageInputType<br>[INFO]: *VersionedInputType<br>[INFO]: *FolderInput<br>[INFO]: *FileFilterInput<br>[INFO]: *FileUsage<br>[INFO]: *DescendantFileCount<br>[INFO]: *PublicationNotice<br>[INFO]: *Folder<br>[INFO]: *File<br>[INFO]: *FileInput<br>[INFO]: (... and 49 more)<br>[INFO]: Types deleted: 0<br>
```

We probably need need to redirect those logs to to STDERR or something for "on demand" request.

# Pull request
- [ ] https://github.com/silverstripe/silverstripe-graphql/pull/448

**Comment by maxime-rainville** [comment]
Part of the problem is that we're using a new [`SilverStripe\GraphQL\Schema\Logger`](https://github.com/silverstripe/silverstripe-graphql/blob/master/src/Schema/Logger.php)

I get the impression this should only be used when we are on the `dev/graphql/schema` route. The rest of the time, we should probably used one of the option defined in [`silverstripe/framwork:_config/logging.yml`](https://github.com/silverstripe/silverstripe-framework/blob/4/_config/logging.yml)?

Maybe, we shouldn't even be using `SilverStripe\GraphQL\Schema\Logger` at all.

I'm thinking we define a new `Psr\Log\LoggerInterface.graphql` injector key and assign it something different base on context. Then we replace all instances of `SilverStripe\GraphQL\Schema\Logger::singleton()` with `Injector::get()->instance(LoggerInterface::class . '.graphql')`

**Comment by emteknetnz** [comment]
Seems fine, why would we not just use the regular logger on the `dev/graphql/schema` route as well? i.e. what is `SilverStripe\GraphQL\Schema\Logger` even required for?

**Comment by michalkleiner** [comment]
@emteknetnz — https://github.com/silverstripe/silverstripe-graphql/pull/422#discussion_r772766433

**Comment by maxime-rainville** [comment]
Thanks @michalkleiner :pray: That's super useful context.

**Comment by emteknetnz** [comment]
okay, given that context, seems like we should just delete the graphql logger and use monolog/logger like everything else?

The 'output to console' use case looks like it could be done like this

```php
class ManifestCacheFactory extends DefaultCacheFactory
{
    public function __construct(array $args = [], LoggerInterface $logger = null)
    {
        // Build default manifest logger
        if (!$logger) {
            $logger = new Logger("manifestcache-log");
            if (Director::isDev()) {
                $logger->pushHandler(new StreamHandler('php://output'));
            } else {
                $logger->pushHandler(new ErrorLogHandler());
            }
        }
...
```

https://github.com/silverstripe/silverstripe-framework/blob/4/src/Core/Cache/ManifestCacheFactory.php#L27

**Comment by emteknetnz** [comment]
:thinking: though @maxime-rainville even with this above ^ you'd still end up with the '[INFO]' in the HTTP response when build "on demand" on your local laptop? (cos the check is Director::isDev())

Maybe we just need to be be checking is the `php_sapi_name() === 'cli' as to whether to log to console/http-response, or to "other"

**Comment by michalkleiner** [comment]
> Maybe we just need to be be checking is the `php_sapi_name() === 'cli' as to whether to log to console/http-response, or to "other"

`Environment::isCli`

**Comment by maxime-rainville** [comment]
The build output is still useful if you're calling the build task in the browser via http://example.com/dev/graphql/build

Got a PR that tries to address the problem by redirecting the logging to the standard Silverstripe CMS error log. https://github.com/silverstripe/silverstripe-graphql/pull/448

**Comment by maxime-rainville** [comment]
Resolved by #448
